### ISSUE TITLE
nn.Conv2d doesn't handle well padding in the borders of matrix

### ISSUE THREAD

**Issue by alejandrodemiquel** [pytorch/pytorch#36424]
## 🐛 Bug

I'm training a simple model with just one convolution with a kernel of size 3. I want the output to have the same shape as the input, so I add padding of 1 with value 0. You can see the model I use below.
I have 3 channels, and I use one-hot-encoding, so I only have ones and zeros in the input.
After training the model, when executing the forward, I find that some of the values in the borders are incorrect.

For example if the input is:

1 1 1 0 0 0
1 1 1 0 0 0
1 1 1 1 1 1
0 0 0 1 1 1

I get as output:

1 2 1 0 0 0
1 2 1 0 0 0
1 1 1 1 1 1
0 0 0 1 1 1

which doesn't make sense, since the element [0,1] of the output should be equal to the element [2,5].

## To Reproduce

See attached file in comment below for repro steps.

## Environment

PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows 10 Home
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.1.168
GPU models and configuration: GPU 0: GeForce MX130
Nvidia driver version: 430.86
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.16.5
[pip] numpydoc==0.9.1
[pip] torch==1.4.0
[pip] torchvision==0.5.0
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.1.243             h74a9793_0  
[conda] mkl                       2019.4                      245  
[conda] mkl-service               2.3.0            py37hb782905_0  
[conda] mkl_fft                   1.0.14           py37h14836fe_0  
[conda] mkl_random                1.1.0            py37h675688f_0  
[conda] numpy                     1.16.5           py37h19fb1c0_0  
[conda] numpy-base                1.16.5           py37hc3f5095_0  
[conda] numpydoc                  0.9.1                      py_0  
[conda] pytorch                   1.4.0           py3.7_cuda101_cudnn7_0    pytorch
[conda] torchvision               0.5.0                py37_cu101    pytorch

**Comment by alejandrodemiquel** [comment]
See the attached file for complete repro steps.

[reproSteps.txt](https://github.com/pytorch/pytorch/files/4464111/reproSteps.txt)

**Comment by ptrblck** [comment]
I'm not sure I understand the use case completely.
Given an input of:
```python
tensor([[[[1., 1., 1., 1., 0., 0., 0., 0.],
          [1., 1., 1., 1., 0., 0., 0., 0.],
          [1., 1., 1., 1., 0., 0., 0., 0.],
          [0., 0., 0., 1., 0., 0., 0., 0.],
          [0., 0., 0., 1., 1., 1., 1., 1.],
          [0., 0., 0., 1., 1., 1., 1., 1.],
          [0., 0., 0., 1., 1., 1., 1., 1.],
          [0., 0., 0., 1., 1., 1., 1., 1.]],

         [[0., 0., 0., 0., 1., 1., 1., 1.],
          [0., 0., 0., 0., 1., 1., 1., 1.],
          [0., 0., 0., 0., 1., 1., 1., 1.],
          [1., 1., 1., 0., 1., 1., 1., 1.],
          [1., 1., 1., 0., 0., 0., 0., 0.],
          [1., 1., 1., 0., 0., 0., 0., 0.],
          [1., 1., 1., 0., 0., 0., 0., 0.],
          [1., 1., 1., 0., 0., 0., 0., 0.]],

         [[0., 0., 0., 0., 0., 0., 0., 0.],
          [0., 0., 0., 0., 0., 0., 0., 0.],
          [0., 0., 0., 0., 0., 0., 0., 0.],
          [0., 0., 0., 0., 0., 0., 0., 0.],
          [0., 0., 0., 0., 0., 0., 0., 0.],
          [0., 0., 0., 0., 0., 0., 0., 0.],
          [0., 0., 0., 0., 0., 0., 0., 0.],
          [0., 0., 0., 0., 0., 0., 0., 0.]]]])
```
the output at `out[0, :, 1, 1]` should have the values `model.conv.weight.sum([2, 3])[:, 0]`, if we disable the bias, which is the case in your example.
Note that each filter uses all input channels to create the activation map.

Also, I think the [forum](https://discuss.pytorch.org/) is a better place for these kind of questions. ;)

**Comment by alejandrodemiquel** [comment]
Ah probably I had a misunderstanding about how biases are handled then. Sorry about that, next time I'll ask  these kind of questions in the forum. Thanks for the help!
